### ISSUE TITLE
Part number

### ISSUE THREAD

**Issue by vlopezalvarez** [ietf-ivy-wg/network-inventory-yang#15]
Hi, I've seen that we have in the draft the concept of "part number". RFC 8348 defines the parameter model-name to obtain this information. 

"The vendor-specific model name identifier string
              associated with this physical component.  The preferred
              value is the customer-visible part number, which may be
              printed on the component itself."

As we're changing the model of 8348. Is there any reason to include the "part number" instead of reusing model-name?

**Comment by italobusi** [comment]
**[2024-01-17 Network Inventory weekly call](https://github.com/ietf-ivy-wg/network-inventory-yang/blob/main/minutes/minutes-2024-01-17.md)**

The rationale is described in section 2.4.3 of the draft

It has been noted that the YANG description needs to be updated

- [ ] @YuChaode update the YANG description for the part-number

- [ ] @sergiobelotti  check offline with Victor if he is ok with the proposed resolution

**Comment by italobusi** [comment]
> **[2024-01-24 Base Network Inventory weekly call](https://github.com/ietf-ivy-wg/network-inventory-yang/blob/main/minutes/minutes-2024-01-24.md)**

Model changes proposed by Chaode has been reviewed during the call

The definition of part-number of the component has been updated based on the discussion

The part-number attribute is not applicable to NE. It has been agreed to rename the attribute as assembly-id as a working name and open a new issue to refine the name and description of the assembly-id attribute: see issue xxx

> * [ ]  @YuChaode update the YANG description for the part-number

**Comment by lana-wu** [comment]
Similar with Victor, I suggest that "model-name" in RFC 8348 be reused. This node applies only to components, not NEs.

**Comment by sergiobelotti** [comment]
I tend to agree with Bo's comment: one thing is discussing the opportunity or not to use part-number on behalf of model-name for component only another thing is mixing things and introduce another term , not defined in RFC8348, considering also NE while model-name does not .

**Comment by italobusi** [comment]
> "The vendor-specific model name identifier string
> associated with this physical component. The preferred
> value is the customer-visible part number, which may be
> printed on the component itself."

@sergiobelotti I am a bit confused about whether your comment referring to:
1. use of "part-number" or "model-name" attribute for the component
2. use of "asset-id" or "model-name" (or "product-name" for the NE

I would suggest to you this issue to discuss point 1 and issue #33 to discuss point 2

@sergiobelotti @lana-wu I have understood from the previous calls that we have agreed to use "part-number" or "model-name" attribute for the component pending confirmation with Victor and marked this issue as agreed accordingly

Please let me know if my understanding is not correct so we can discuss it again in the next calls (and mark this issue as under-discussion to avoid confusion)

**Comment by lana-wu** [comment]
@italobusi @sergiobelotti 
 I agree with the first point, instead of a new name "assembly-id“. 
For the second point, I am not sure about "asset-id", which is NOT a vendor-specific but a customer defined tag.

**Comment by sergiobelotti** [comment]
> > "The vendor-specific model name identifier string
> > associated with this physical component. The preferred
> > value is the customer-visible part number, which may be
> > printed on the component itself."
> 
> @sergiobelotti @lana-wu I am a bit confused about whether your comments are referring to:
> 
> 1. use of "part-number" or "model-name" attribute for the component
> 2. use of "asset-id" or "model-name" (or "product-name" for the NE
> 
> I would suggest using this issue to discuss point 1 and issue #33 to discuss point 2

@italobusi: my comment is related to the above comment from Bo  https://github.com/ietf-ivy-wg/network-inventory-yang/issues/15#issuecomment-1918268011, and is referring to the first comment from Victor regarding modul-name vs. part-number.  I agree on your suggestion to separate the two issues .
> 
> Regarding point 1, I have understood from the previous calls that we have agreed to use "part-number" or "model-name" attribute for the component, pending confirmation with Victor and marked this issue as agreed accordingly
> 
> Please let me know if my understanding is not correct: if needed, we can discuss it again in the next calls (and mark this issue as under-discussion to avoid confusion)
@italobusi : yes you understanding is correct

**Comment by italobusi** [comment]
> Part-number is widely used in the community. Besides it matches with openconfig part-no, so industry-wide there is no confussion.

Actually I noted that OpenConfig defines both part number and model name attributes:

https://github.com/openconfig/public/blob/master/release/models/platform/openconfig-platform.yang

Since the model name is already defined in RFC8348, an alternative solution could be to define:

```
  part-number as “The system-assigned and vendor-specific part number of the component”
  product-name as “The system-assigned, vendor-specific and human-readable string describing the component”
```

Following the similar approach, we can also define:

```
  serial-number as “The system-assigned and vendor-specific serial number of the component”
  asset-id as “The user-assigned asset tracking identifier of the component”
```

**Comment by italobusi** [comment]
**[2024-02-21 Base Network Inventory weekly call](https://demo.hedgedoc.org/cQyswfIUT4WbnkvvbvjQ1Q)**

The following attributes have been agreed:

- part-number as "The vendor-specific part number of the component type. It is expected that vendors assign unique part numbers to different component types within the scope of the vendor."

- product-name as "The vendor-specific and human-interpretable string describing the component type. It is expected that vendors assign unique product names to different component types within the scope of the vendor."

- serial-number as "The vendor-specific serial number of the component instance. It is expected that vendors assign unique serial numbers to different component instances within the scope of the part number."

- [ ] @YuChaode : update the draft and the model accordingly

**Comment by italobusi** [comment]
**[2024-03-06 Base Network Inventory weekly call](https://github.com/ietf-ivy-wg/network-inventory-yang/blob/main/minutes/minutes-2024-03-06.md)**

The YANG descriptions in the I-D version uploaded for IETF 119 are not fully aligned with the descriptions agreed during the call on February 21, 2024

- [ ] @YuChaode update the definitions in the YANG model to align with the definitions in [2024-02-21 Base Network Inventory weekly call](https://github.com/ietf-ivy-wg/network-inventory-yang/blob/main/minutes/minutes-2024-02-21.md)

**Comment by italobusi** [comment]
Closed with 683ea0729a10a841b3fa7de3ec57eb8f33920e0c
